### PR TITLE
[POC] show the reason why the debugger is paused

### PR DESCRIPTION
**Pull request by clarkbw**
Just a proof of concept for now, if anyone wants to take this further feel free to pick this up.  I think we want input from @helenvholmes first.  I'm not sure the sidebar is actually a good location, the way it pushes everything down doesn't seem like the best design.  A good path forward could be to move the logic out of the component and into a util function such that we can try a couple different variants of the components.

### Summary of Changes

* Create a new component that displays why the debugger is paused

### Test Plan

Tell us a little a bit about how you tested your patch.

Example test plan:

- [x] Pause on breakpoint
- [x] Pause on debugger statement
- [x] Turn off feature flag (no element rendered)
- [x] Turn on feature flag (element rendered)
- [ ] Pause on exception
- [ ] Pause on ...

### Screenshots/Videos

screen shot 2016-11-29 at 8 02 50 am

screen shot 2016-11-29 at 8 02 46 am

### CODE REVIEW

**Comment by jasonLaster** [comment]
Thanks @clarkbw this is a cool idea and a great thing we can experiment with in 2017.

Could you put it behind a feature flag so that we can setup some experiments.

File: src/components/WhyPaused.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    pauseInfo: ImPropTypes.map
+  },
+
+  displayName: "WhyPaused",
+
+  render() {
+    const { pauseInfo } = this.props;
+    const reason = getPauseReason(pauseInfo);
+
+    return pauseInfo ?
```
Comment (jasonLaster) [inline src/components/WhyPaused.js]: perhaps if we don't have a reason we don't show the highlight for now?

Comment (clarkbw) [inline src/components/WhyPaused.js]: Good point, better to be quiet about what we are unsure about.

**Comment by jasonLaster** [comment]
Really cool! 

Lets make sure we can schedule the design + UR time to setup this experiment. In Q1

**Pull request merged**